cmd/update/update_scheduler.go

```diff
@@ -0,0 +1,157 @@
+// maestro-cli
+// https://github.com/topfreegames/maestro-cli
+//
+// Licensed under the MIT license:
+// http://www.opensource.org/licenses/mit-license
+// Copyright © 2017 Top Free Games <backend@tfgco.com>
+
+package update
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"github.com/spf13/cobra"
+	"github.com/topfreegames/maestro-cli/common"
+	"github.com/topfreegames/maestro-cli/extensions"
+	"github.com/topfreegames/maestro-cli/interfaces"
+	"google.golang.org/protobuf/encoding/protojson"
+
+	v1 "github.com/topfreegames/maestro/pkg/api/v1"
+	yaml "gopkg.in/yaml.v2"
+	k8s_yaml "sigs.k8s.io/yaml"
+)
+
+var marshler = &runtime.HTTPBodyMarshaler{
+	Marshaler: &runtime.JSONPb{
+		MarshalOptions: protojson.MarshalOptions{
+			EmitUnpopulated: true,
+		},
+	},
+}
+
+// updateSchedulerCmd represents the update command
+var updateSchedulerCmd = &cobra.Command{
+	Use:     "scheduler",
+	Short:   "Updates scheduler",
+	Example: "maestro-cli update scheduler ./scheduler.yaml",
+	Long:    "Uses a file (argument) to update a new scheduler on Maestro.",
+	Args:    validateArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, config, err := common.GetClientAndConfig()
+		if err != nil {
+			return err
+		}
+
+		return NewUpdateScheduler(client, config).run(cmd, args)
+	},
+}
+
+type UpdateScheduler struct {
+	client interfaces.Client
+	config *extensions.Config
+}
+
+func NewUpdateScheduler(client interfaces.Client, config *extensions.Config) *UpdateScheduler {
+	return &UpdateScheduler{
+		client: client,
+		config: config,
+	}
+}
+
+func validateArgs(_ *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		return errors.New("missing arg with scheduler config file path")
+	}
+
+	filePath := args[0]
+	_, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return fmt.Errorf("error while reading file: %w", err)
+	}
+
+	return nil
+}
+
+func (cs *UpdateScheduler) run(_ *cobra.Command, args []string) error {
```

Comment:
@caiordjesus  don't you think this function is too big? couldn't it be split better to be more readable?
I understood that code inside the 'for' executes the most important part of the flow. maybe this could be in a separate function.

Reply:
Great insight! Thanks!
Just did it!